functional-tests/tests/test_analyzer.py

```diff
@@ -11,9 +10,91 @@ def test_given_a_correct_analyze_input_then_return_full_response():
+
+
+@pytest.mark.api
+def test_given_analyze_threshold_input_then_return_result_above_threshold():
+    request_body = {
+        "text": "John Smith drivers license is AC432223", "language": "en", "score_threshold": 0.7}
+
+    response_status, response_content = analyze(request_body)
+
+    assert response_status == 200
+    assert response_content == ["{'entity_type': 'PERSON', 'start': 0, 'end': 10, 'score': 0.85, 'analysis_explanation': {'recognizer': 'SpacyRecognizer', 'pattern_name': None, 'pattern': None, 'original_score': 0.85, 'score': 0.85, 'textual_explanation': \"Identified as PERSON by Spacy's Named Entity Recognition\", 'score_context_improvement': 0, 'supportive_context_word': '', 'validation_result': None}}"]
+
+
+@pytest.mark.api
+def test_given_no_analyze_text_input_then_return_error():
+    request_body = {}
+
+    response_status, response_content = analyze(request_body)
+
+    assert response_status == 500
+    assert response_content == {"error": "No language provided"}
+
+
+@pytest.mark.api
+def test_given_no_analyze_language_input_then_return_error():
+    request_body = {"language": "en"}
+
+    response_status, response_content = analyze(request_body)
+
+    assert response_status == 500
+    assert response_content == {"error": "No text provided"}
+
+
+@pytest.mark.api
+def test_given_no_analyze_text_input_then_return_error():
+    request_body = {
+        "text": "John Smith drivers license is AC432223"}
+
+    response_status, response_content = analyze(request_body)
+
+    assert response_status == 500
+    assert response_content == {"error": "No language provided"}
+
+
+@pytest.mark.api
+def test_given_a_incorrect_analyze_language_input_then_return_error():
+    request_body = {
+        "text": "John Smith drivers license is AC432223", "language": "zz"}
+
+    response_status, response_content = analyze(request_body)
+
+    assert response_status == 500
+    assert response_content == {"error": "No matching recognizers were found to serve the request."}
+
+
+@pytest.mark.api
+def test_given_a_correlationid_analyze_input_then_return_normal_response():
+    request_body = {
+        "text": "John Smith drivers license is AC432223", "language": "en", "correlation_id": "123"}
+
+    response_status, response_content = analyze(request_body)
+
+    assert response_status == 200
+
+
+@pytest.mark.api
+def test_given_a_trace_true_analyze_input_then_return_normal_response():
+    request_body = {
+        "text": "John Smith drivers license is AC432223", "language": "en", "trace": "1"}
+
+    response_status, response_content = analyze(request_body)
+
+    assert response_status == 200
+
+
+def test_given_analyze_entities_input_then_return_results_only_with_those_entities():
```

Comment:
Are we testing in case the entity type is not in the text?
Are we testing a case where there's no PII in the text?

Reply:
We are not testing the validity of the anlyzer engine, as it has it's own tests.
but rather that the http passes the values correctly.
So I think it's safe.
In addition to that the engine simply filters out results, regardles if there are PIIs or Entities not in text.